miniscope_io/stream_daq.py

```diff
@@ -159,22 +162,34 @@ def _parse_header(self, buffer: bytes) -> Tuple[StreamBufferHeader, np.ndarray]:
 
         return header_data, payload
 
-    def _trim(self, data: np.ndarray, expected_size: int, logger: logging.Logger) -> np.ndarray:
+    def _trim(
+        self,
+        data: np.ndarray,
+        expected_size_array: List[int],
+        header: StreamBufferHeader,
+        logger: logging.Logger,
+    ) -> np.ndarray:
         """
         Trim or pad an array to match an expected size
         """
-        if data.shape[0] != expected_size:
+        expected_payload_size = expected_size_array[0]
+        expected_data_size = expected_size_array[header.frame_buffer_count]
```

Comment:
Just to let you know:
- The buffer sizes are fixed because it's not really worth making that adaptive.
- Thus, the final buffer in the frame isn't filled with pixels so it needs to be treated slightly differently.

And some logs before making this change and comments are here:
#36